### PR TITLE
Streamline the code style in containers/

### PR DESCRIPTION
**Pull request by pdostal**
Hello,

this is a follow-up after #11828 

 * Use `test_containered_app()` when possible
 * Merge `Dockerfile` and `Dockerfile.python`
 * Use `} else {` instead of `}\n else {`
 * Subroutine aceepts named parameters
 * Die when missing required parameter

- Verification run: [SLE15-SP2](http://pdostal-server.suse.cz/tests/11075) [Tumbleweed](http://pdostal-server.suse.cz/tests/11086)

### CODE REVIEW

File: lib/containers/container_images.pm [hunk 1]
```diff
@@ -53,10 +54,22 @@ sub build_container_image {
...
+
+    die "You must define the runtime!"    unless $runtime;
+    die "You must define the Dockerfile!" unless $dockerfile;
+
     my $dir = "/root/containerapp";
-    set_up($dir, "Dockerfile.python");
-    build_img($dir, $runtime);
+
+    # Setup the environment
+    set_up("$dir", $dockerfile);
```
Comment (grisu48) [inline lib/containers/container_images.pm:67]: One could argue, that `set_up` is a not really descriptive function name and could be renamed to `container_set_up` or something similar.

File: tests/containers/buildah_image.pm [hunk 2]
```diff
@@ -26,23 +26,23 @@ use version_utils qw(get_os_release check_os_release);
...
         if (check_os_release('suse', 'PRETTY_NAME')) {
             # sle15-working-container is the default name given to a container. it is created in test_container_image
-            test_opensuse_based_image(image => 'sle15-working-container', runtime => $runtime);
+            test_opensuse_based_image(image => 'sle15-working-container', runtime => 'buildah');
             allow_selected_insecure_registries(runtime => 'podman');
-            test_containered_app('podman', $iname);
         }
     }
     scc_restore_docker_image_credentials();
     clean_container_host(runtime => 'podman');
```
Comment (grisu48) [inline tests/containers/buildah_image.pm:44]: Is this still required here?

**Review by grisu48 — APPROVED** [review]
Like it! Two minor comments, other than that LGTM :rocket:

File: tests/containers/buildah_image.pm (hunk 2 above)
Comment (grisu48) [inline tests/containers/buildah_image.pm:44]: Is this still requiredin `buildah_image.pm`?

Comment (grisu48) [reply tests/containers/buildah_image.pm:44]: Readded the comment, because it got accidentally deleted.

Comment (pdostal) [reply tests/containers/buildah_image.pm:44]: Yes, in this module both `buildah` and `podman` produce images.

**Review by punkioudi — APPROVED** [review]
lgtm Pavel!

**Review by jlausuch — APPROVED** [review]
Nice refactor. Good job

**Review by ggkioulis — APPROVED** [review]
Great work Pavel!

**Pull request merged**